### PR TITLE
Use FullscreenGrid for everything

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/652

**TODO**: the file tree area is an improvement over master but I'd like to improve it more in https://github.com/mozilla/addons-code-manager/issues/653

Notes:
- I made the Navbar a bit smaller since the header is always present
- I removed borders from some things since the grid panels have borders now
- Everything else looks more or less the same except all screens are in the grid
- I adapted `FullscreenGrid` to make it easier to work with for our specific app

<details>
<summary>Screencast</summary>

</details>

<details>
<summary>Screenshots</summary>

acm_panels_initial
acm_panels_logging_in
acm_panels_loading
acm_panels_browse
acm_panels_compare
acm_panels_errors

</details>

### CODE REVIEW

**Comment by kumar303** [comment]
@bobsilverberg I rebased this to get the linter message support in the overview. Clicking on a line always scrolls to the exact line like magic!

File: src/components/FullscreenGrid/index.tsx [hunk 1]
```diff
@@ -3,38 +3,63 @@ import makeClassName from 'classnames';
...
+      <aside
+        className={makeClassName(styles.mainSidePanel, mainSidePanelClass)}
+      >
+        {mainSidePanel}
+      </aside>
+      <main className={makeClassName(styles.content, className)}>
+        {children}
+      </main>
+      <aside className={makeClassName(styles.altSidePanel, altSidePanelClass)}>
+        {altSidePanel}
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/index.tsx]: So if `altSidePanel` is undefined then it would show an empty panel? There's no current way to hide the `aside` for this `altSidePanel`?

File: src/components/FullscreenGrid/styles.module.scss [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-$grid-gap: 10px;
-
 .FullscreenGrid {
   display: grid;
-  grid-gap: $grid-gap;
+  grid-gap: $default-padding;
   grid-template-areas:
     'FGHeader FGHeader FGHeader'
-    'FGNavPanel FGContentPanel FGAltPanel'
+    'FGMainSidePanel FGContent FGAltSidePanel'
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/styles.module.scss]: What's "FG" stand for?

File: src/pages/Compare/index.spec.tsx [hunk 3]
```diff
@@ -260,7 +261,10 @@ describe(__filename, () => {
 
     const root = render({ store });
 
-    expect(root.find(`.${styles.error}`)).toHaveLength(2);
+    expect(
+      getContentShellPanel(root, 'mainSidePanel').find(`.${styles.error}`),
```
Comment (bobsilverberg) [inline src/pages/Compare/index.spec.tsx]: Shouldn't this error be displayed in the `Header`, rather than in the `mainSidePanel`?

File: src/test-helpers.tsx [hunk 4]
```diff
@@ -590,3 +592,15 @@ export const simulateLinterProvider = (
     selectedMessageMap,
   });
 };
+
+/*
+ * Given a ShallowWrapper for a component that renders ContentShell,
+ * return a wrapper around one of its panel attributes.
+ */
+export const getContentShellPanel = (
```
Comment (bobsilverberg) [inline src/test-helpers.tsx]: Nice helper! 👍

File: stories/CodeView.stories.tsx [hunk 5]
```diff
@@ -56,7 +56,12 @@ const render = ({
     version: createInternalVersion(fakeVersion),
     ...moreProps,
   };
-  return renderWithStoreAndRouter(<CodeView {...props} />, store);
+  return renderWithStoreAndRouter(
+    <div className="CodeView-panel">
```
Comment (bobsilverberg) [inline stories/CodeView.stories.tsx]: Why is this needed?

File: src/components/FullscreenGrid/index.tsx [hunk 6]
```diff
@@ -3,38 +3,63 @@ import makeClassName from 'classnames';
 
 import styles from './styles.module.scss';
 
-type PanelProps = {
-  children: React.ReactNode | React.ReactNode[];
-  className?: string;
-};
+type AnyReactNode = React.ReactNode | React.ReactNode[];
```
[the remark this answers is not in the document]
Comment (kumar303) [reply src/components/FullscreenGrid/index.tsx]: 'Any react node' seems like a way of saying 'a single react node or an array of react nodes' to me. In JSX, there isn't much difference between a single node or an array of nodes.

**Comment by kumar303** [comment]
> My main question is about the right-hand side panel. There seems to be no way to hide it, even if there is no content. This is particulary evident on the `Compare` page, which, AFAICT doesn't support having a `CodeOverview`. Do we plan to implement a `CodeOverview` to work with the `Compare` page? 

Definitely! I thought I'd filed but I guess not. Filed: https://github.com/mozilla/addons-code-manager/issues/672

> Do we want to be able to hide the side panel when it serves no purpose?

I can't really think of when the side panel won't be in use aside from loading screens. We could hide it on the loading screens but I think showing an empty outline is fine, too.

File: src/components/FullscreenGrid/styles.module.scss (hunk 2 above)
Comment (kumar303) [reply src/components/FullscreenGrid/styles.module.scss]: It stands for `FullscreenGrid`. As I understand it, `grid-template-area` names are global to all CSS. I wanted to protect this namespace a little bit but I didn't want to prefix it with `FullscreenGrid` because that's too long. The longer these names become, the harder it is to see the pattern of columns and rows.

File: src/pages/Compare/index.spec.tsx (hunk 3 above)
Comment (kumar303) [reply src/pages/Compare/index.spec.tsx]: This component is far removed from where the `Header` is rendered so it would be tricky to do that. Also, I think it's still readable to see an error in this panel.

File: stories/CodeView.stories.tsx (hunk 5 above)
Comment (kumar303) [reply stories/CodeView.stories.tsx]: `CodeView` no longer renders its own border. This makes it hard to tell where the title of the story ends and the component content begins. I added a border around the story content which highlights the component's container.

**Comment by kumar303** [comment]
@bobsilverberg thanks for looking! Let me know if you have further questions.
